public/app/plugins/datasource/influxdb/components/editor/query/influxql/visual/VisualInfluxQLEditor.tsx

```diff
@@ -54,6 +54,16 @@ export const VisualInfluxQLEditor = (props: Props): JSX.Element => {
   const { measurement, policy } = query;
   const { retentionPolicies } = useRetentionPolicies(datasource);
 
+  useEffect(() => {
+    if (!policy) {
+      props.onChange({
+        ...query,
+        policy: retentionPolicies[0],
+      });
+      props.onRunQuery();
+    }
+  }, [policy, props, query, retentionPolicies]);
+
```

Comment:
If there is no retention policy (you just add a new query row), set back that default policy into the query model